src/Lean/Elab/Tactic/BVAckermannize.lean

```diff
@@ -0,0 +1,446 @@
+/-
+Copyright (c) 2024 Siddharth Bhat. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Siddharth Bhat
+
+This file implements strict ackermannization [1, 2]
+
+[1] https://lara.epfl.ch/w/_media/model-based.pdf
+[2] https://leodemoura.github.io/files/oregon08.pdf
+[3] https://github.com/Z3Prover/z3/blob/d047b86439ec209446d211f0f6b251ebfba070d8/src/ackermannization/lackr.cpp#L206
+[4] https://github.com/Z3Prover/z3/blob/d047b86439ec209446d211f0f6b251ebfba070d8/src/ackermannization/lackr_model_constructor.cpp#L344
+-/
+prelude
+import Lean.Expr
+import Lean.Message
+import Std.Tactic.BVDecide.Bitblast
+import Std.Tactic.BVAckermannize.Syntax
+import Lean.Meta.Basic
+import Lean.Elab.Tactic.Basic
+import Lean.Meta.LitValues
+import Lean.Meta.InferType
+import Lean.Elab.Tactic.FalseOrByContra
+import Lean.Meta.Tactic.Assert
+import Lean.Meta.Tactic.Util
+import Lean.Meta.Tactic.Congr
+import Lean.Meta.Tactic.Replace
+
+open Lean Elab Meta Tactic
+
+initialize Lean.registerTraceClass `Meta.Tactic.bv_ack
+
+namespace Ack
+
+ /-- Types that can be bitblasted by bv_decide. -/
+ inductive BVTy
+   /-- Booleans. -/
+   | Bool
+   /-- Bitvectors of a fixed width `w`. -/
+   | BitVec (w : Nat)
+   deriving Hashable, DecidableEq, Inhabited
```

Comment:
```suggestion
 inductive BVTy
   /-- Booleans. -/
   | bool
   /-- Bitvectors of a fixed width `w`. -/
   | bitVec (w : Nat)
   deriving Hashable, DecidableEq, Inhabited
```

I'm not sure if this is a Mathlib-specific convention, but I'd expect the constructors of an inductive to follow snakeCase naming.

In this case, specifically, I see where you get the capitals from, but feel the similarity between `.Bool : BVTy` and  `Bool : Type` might cause unnecessary confusion as compared to `.bool : BVTy`.

Reply:
Done